pkg/syncer/metadatasyncer.go

```diff
@@ -1034,8 +1067,16 @@ func csiUpdatePod(ctx context.Context, pod *v1.Pod, metadataSyncer *metadataSync
 				containerCluster := cnsvsphere.GetContainerCluster(metadataSyncer.configInfo.Cfg.Global.ClusterID, metadataSyncer.configInfo.Cfg.VirtualCenter[metadataSyncer.host].User, metadataSyncer.clusterFlavor)
 				var volumeHandle string
 				var err error
-				if metadataSyncer.coCommonInterface.IsFSSEnabled(ctx, common.CSIMigration) && pv.Spec.VsphereVolume != nil {
+				if pv.Spec.VsphereVolume != nil && metadataSyncer.coCommonInterface.IsFSSEnabled(ctx, common.CSIMigration) {
 					migrationVolumeSpec := migration.VolumeSpec{VolumePath: pv.Spec.VsphereVolume.VolumePath, StoragePolicyName: pv.Spec.VsphereVolume.StoragePolicyName}
+					if volumeMigrationService == nil {
```

Comment:
We are repeating the following code block, at 5 locations. Can we wrap this into the util function for example `initvolumeMigrationService(ctx)`? 

```
if volumeMigrationService == nil {
    var err error
    volumeMigrationService, err = migration.GetVolumeMigrationService(ctx, &metadataSyncer.volumeManager, metadataSyncer.configInfo.Cfg)
    if err != nil {
		log.Errorf("failed to get migration service. Err: %v", err)
		return
    }
}
```

Reply:
Done